### PR TITLE
[Move-package] Enable bytecode as dependencies

### PR DESCRIPTION
**Pull request by rahxephon89**
## Motivation

Re-enable the feature to use bytecode as dependencies when compiling Move programs.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#developer-workflow)?

Yes

## Test Plan

Add later

### CODE REVIEW

**Comment by rahxephon89** [comment]
Hi @tnowacki @tzakian, although it will not be landed on main, could you take a look and give some feedback? Thanks!

File: language/move-compiler/src/parser/mod.rs [hunk 1]
```diff
@@ -63,6 +63,8 @@ pub(crate) fn parse_program(
 
     let targets = find_move_filenames_with_address_mapping(targets)?;
     let mut deps = find_move_filenames_with_address_mapping(deps)?;
+    // Remove bytecode files
+    deps.retain(|p| !p.path.as_str().ends_with(MOVE_COMPILED_EXTENSION));
```
Comment (tnowacki) [inline language/move-compiler/src/parser/mod.rs:67]: Could you explain the reasoning to remove bytecode files from deps? 

Targets means that the compiler is intended to produce bytecode modules for those sources files, which I'm not sure makes sense here

Comment (rahxephon89) [reply language/move-compiler/src/parser/mod.rs:67]: At this point, deps may already contain paths to mv files that are used as external dependencies, which will cause panic later when calling the function parse_file.

Comment (tnowacki) [reply language/move-compiler/src/parser/mod.rs:67]: So the "interface" files have been generated at this point right? Would it make sense to just remove the `.mv` files from deps after generating the source files for them?

Comment (rahxephon89) [reply language/move-compiler/src/parser/mod.rs]: Thanks for the advice Todd! I moved the code to [here]( https://github.com/move-language/move/blob/4d24d3df379cf655eb43ef7fccd7db3e71472860/language/move-compiler/src/command_line/compiler.rs#L611). I am wondering whether it makes more sense?

**Review by wrwg — APPROVED** [review]

**Pull request merged**